atomic_reactor/plugins/exit_sendmail.py

```diff
@@ -0,0 +1,263 @@
+"""
+Copyright (c) 2015 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+from email.mime.text import MIMEText
+import os
+import smtplib
+import socket
+try:
+    from urlparse import urljoin
+except ImportError:
+    from urllib.parse import urljoin
+
+from dockerfile_parse import DockerfileParser
+import requests
+
+from atomic_reactor.plugin import ExitPlugin, PluginFailedException
+from atomic_reactor.plugins.pre_check_and_set_rebuild import is_rebuild
+from atomic_reactor.source import GitSource
+
+
+class SendMailPlugin(ExitPlugin):
+    """This plugins sends notifications about build results.
+
+    Example configuration (see arguments for init for detailed explanation):
+        "exit_plugins": [{
+                "name": "sendmail",
+                "args": {
+                    "send_on": ["auto_canceled", "auto_fail"],
+                    "url": "https://openshift-instance.com",
+                    "pdc_url": "https://pdc-instance.com",
+                    # pdc_secret_path is filled in automatically by osbs-client
+                    "pdc_secret_path": "/path/to/file/with/pdc/token",
+                    "smtp_url": "smtp-server.com",
+                    "from_address": "osbs@mycompany.com",
+                    "error_addresses": ["admin@mycompany.com"],
+                    # optional arguments follow
+                    "pdc_verify_cert": true,
+                    "pdc_component_df_label": "BZComponent",
+                    "pdc_contact_role": "Build_Owner"
+                }
+        }]
+    """
+    key = "sendmail"
+
+    # symbolic constants for states
+    MANUAL_SUCCESS = 'manual_success'
+    MANUAL_FAIL = 'manual_fail'
+    AUTO_SUCCESS = 'auto_success'
+    AUTO_FAIL = 'auto_fail'
+    AUTO_CANCELED = 'auto_canceled'
+
+    allowed_states = set([MANUAL_SUCCESS, MANUAL_FAIL, AUTO_SUCCESS, AUTO_FAIL, AUTO_CANCELED])
+
+    PDC_TOKEN_FILE = 'pdc.token'
+
+    def __init__(self, tasker, workflow, send_on=None, url=None, pdc_url=None,
+                 pdc_verify_cert=True, pdc_component_df_label="BZComponent", pdc_secret_path=None,
+                 pdc_contact_role="Build_Owner", smtp_url=None, from_address=None,
+                 error_addresses=None):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param url: URL to OSv3 instance where the build logs are stored
+        :param send_on: list of build states when a notification should be sent
```

Comment:
Would be nice to list `url` and `send_on` in the order they appear in the keyword list.